Source/Css/ExCssQuery.cs

```diff
@@ -0,0 +1,304 @@
+            }
+
+            return ops.NthChild(step, offset);
+        }
+
+        private static int GetStep(ChildSelector selector)
+        {
+            var result = selector.Step;
+            if (result == 0)
+            {
+                result = 1;
+            }
+
+            return result;
+        }
+
+        private static Func<IEnumerable<SvgElement>, IEnumerable<SvgElement>> GetFunc(
+            LastChildSelector selector,
+            ExSvgElementOps ops)
+        {
+            var step = GetStep(selector);
+            var offset = selector.Offset;
+
+            if (offset == 0)
+            {
+                return ops.LastChild();
+            }
+
+            return ops.NthLastChild(step, offset);
+        }
+
+        private static Func<IEnumerable<SvgElement>, IEnumerable<SvgElement>> GetFunc(
+            ListSelector listSelector,
+            ExSvgElementOps ops,
+            Func<IEnumerable<SvgElement>,
+                IEnumerable<SvgElement>> inFunc)
+        {
+            List<Func<IEnumerable<SvgElement>, IEnumerable<SvgElement>>> results = new();
+
+            foreach (var selector in listSelector)
+            {
+                results.Add(GetFunc(selector, ops, null));
+            }
+
+            return f =>
+            {
+                var svgElements = inFunc(f);
+                var nodes = results[0](svgElements);
+                for (int i = 1; i < results.Count; i++)
+                {
+                    nodes = nodes.Union(results[i](svgElements));
+                }
+
+                return nodes;
+            };
+        }
+
+        private static Func<IEnumerable<SvgElement>, IEnumerable<SvgElement>> GetFunc(
+            PseudoClassSelector selector,
+            ExSvgElementOps ops,
+            Func<IEnumerable<SvgElement>,
+                IEnumerable<SvgElement>> inFunc)
+        {
+            Func<IEnumerable<SvgElement>, IEnumerable<SvgElement>> pseudoFunc;
+            if (selector.Class == PseudoClassNames.FirstChild)
+            {
+                pseudoFunc = ops.FirstChild();
+            }
+            else if (selector.Class == PseudoClassNames.LastChild)
+            {
+                pseudoFunc = ops.LastChild();
+            }
+            else if (selector.Class == PseudoClassNames.Empty)
+            {
+                pseudoFunc = ops.Empty();
+            }
+            else if (selector.Class == PseudoClassNames.OnlyChild)
+            {
+                pseudoFunc = ops.OnlyChild();
+            }
+            else
+            {
+                if (selector.Class.StartsWith(PseudoClassNames.Not))
+                {
+                    var sel = selector.Class.Substring(PseudoClassNames.Not.Length + 1, selector.Class.Length - 2 - PseudoClassNames.Not.Length);
+                    var parser = new StylesheetParser(true, true, tolerateInvalidValues: true);
+                    var styleSheet = parser.Parse(sel);
```

Comment:
excss does not fully parse the not pseudofunc so I have to parse it myself here.